File: Python/1305. All Elements in Two Binary Search Trees.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getAllElements(self, root1: TreeNode, root2: TreeNode) -> List[int]:
        l = []

        def dfs(node):
            if node: l.append(node.val)
            else: return

            if node.left: dfs(node.left)
            if node.right: dfs(node.right)

        dfs(root1)
        dfs(root2)

        return sorted(l)

File: C++/1305. All Elements in Two Binary Search Trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> l;

    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        dfs(root1);
        dfs(root2);

        sort(l.begin(), l.end());

        return l;
    }

    void dfs(TreeNode* node) {
        if(node) l.push_back(node->val);
        else return;

        if(node->left) dfs(node->left);
        if(node->right) dfs(node->right);
    }
};
